### PR TITLE
add org.tuxemon.tuxemon

### PR DESCRIPTION
**Pull request by Murmele**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

@bitcraft can you have a look on it?

### CODE REVIEW

**Comment by Murmele** [comment]
bot, build org.tuxemon.tuxemon

**Comment by Murmele** [comment]
bot, build org.tuxemon.tuxemon.yaml

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by Murmele** [comment]
@bitcraft do you know why the website tuxemon.org is down?

File: org.tuxemon.Tuxemon.yaml
Comment (Mailaender) [inline org.tuxemon.Tuxemon.yaml:18]: Maybe use https://github.com/flathub/shared-modules/blob/master/pygame/pygame-1.9.6.json instead as it contains PortMidi.

Comment (bitcraft) [reply org.tuxemon.Tuxemon.yaml:18]: I don't believe that we support pygame 1.9.6.  Its very old at this point and all testing is done on pygame versions newer thatn 2.0.0.  Tuxemon doesn't use midi, so if there is a build without portmidi, then that can be used.

**Comment by bitcraft** [comment]
@Murmele are you going to be the maintainer of the flathub releases?

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml:18]: I will check it out.

**Comment by Murmele** [comment]
> @Murmele are you going to be the maintainer of the flathub releases?

Yes I can maintain it

**Comment by bitcraft** [comment]
sounds good.  i don't know of anyone who would be willing or have time to maintain the flathub stuff, so if you are willing to own it, then i don't see a problem

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml:12]: changed. Thanks

**Comment by Murmele** [comment]
Problem:
image

**Comment by bitcraft** [comment]
the master branch is well maintained.  i will merge the last release into it.

File: org.tuxemon.Tuxemon.yaml
Comment (hfiguiere) [inline org.tuxemon.Tuxemon.yaml:42]:
```suggestion
```
this is redundant.

File: org.tuxemon.Tuxemon.yaml
Comment (hfiguiere) [inline org.tuxemon.Tuxemon.yaml:51]:
```suggestion
      - install -Dm755 org.tuxemon.Tuxemon -t /app/bin
```

And then you can remove line 43.

File: org.tuxemon.Tuxemon.yaml
Comment (hfiguiere) [inline org.tuxemon.Tuxemon.yaml:56]: the lack of 128x128 (or scalable) icon will be a problem.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.tuxemon.Tuxemon.yaml
Comment (hfiguiere) [inline org.tuxemon.Tuxemon.yaml:9]: does it support wayland?

File: org.tuxemon.Tuxemon.yaml
Comment (hfiguiere) [inline org.tuxemon.Tuxemon.yaml:17]: does it need access to gamepads?

**Comment by hfiguiere** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by hfiguiere** [comment]
current failure. Appdata failed to validate:
```
builddir/files/share/appdata/org.tuxemon.Tuxemon.appdata.xml: FAILED:
• style-invalid         : <summary> is too short [Tuxemon] minimum is 8 chars
Validation of files failed
```

File: org.tuxemon.Tuxemon.yaml
Comment (bitcraft) [reply org.tuxemon.Tuxemon.yaml:17]: yes

Comment (hfiguiere) [reply org.tuxemon.Tuxemon.yaml:17]: then it probably needs `--device=all`

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml:56]: @bitcraft do you have a 128x128 pixel icon?

**Comment by Murmele** [comment]
> 

There is a pending pullrequest

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml:42]: removed

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml:51]: changed

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml:17]: changed

File: org.tuxemon.Tuxemon.yaml
Comment (bitcraft) [reply org.tuxemon.Tuxemon.yaml]: no, but i will ask around for suggestions

**Comment by barthalion** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by barthalion** [comment]
It's fine to patch the appdata here until a new release is out.

**Comment by Murmele** [comment]
I try to get the appdata into the official repository then I will use that as patch here.

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml]: I did not find a way to disable midi

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

File: org.tuxemon.Tuxemon.yaml
Comment (Murmele) [reply org.tuxemon.Tuxemon.yaml]: @bitcraft should I create a temporary 128x128 icon? I created a patch for the pending pullrequest on Tuxemon. This will be removed, once the mr is merged and a new version of Tuxemon is created

**Comment by barthalion** [comment]
The current error usually means that there's an ID mismatch somewhere, or icon/desktop files are not correctly named. I start from looking into the file in `/app/share/app-info`, check if IDs are correct there, then look for the desktop file, check the `Icon` field, and finally check if that icon is actually present.

**Comment by Murmele** [comment]
> 

Hi @barthalion,

currently there does not exist an 128x128 icon. I'm in contact with the project developers to find a solution

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by Murmele** [comment]
@barthalion Is there a way to do a binary diff?

Otherwise we have to push the changes to the Tuxemon repository and instead of the tag we use than the new commit id.

**Comment by Murmele** [comment]
> > @barthalion Is there a way to do a binary diff?
> > Otherwise we have to push the changes to the Tuxemon repository and instead of the tag we use than the new commit id.
> 
> `use-git: true` as per the manual.
> 
> But to add an icon you might as well add it to the repository here.

Thank you for the response. The icon is in the development branch of Tuxemon, but not in version 0.4.33 so I have to wait until a new version is released.

Where should I add this command?

**Comment by hfiguiere** [comment]
> > > @barthalion Is there a way to do a binary diff?
> > > Otherwise we have to push the changes to the Tuxemon repository and instead of the tag we use than the new commit id.
> > 
> > 
> > `use-git: true` as per the manual.
> > But to add an icon you might as well add it to the repository here.
> 
> Thank you for the response. The icon is in the development branch of Tuxemon, but not in version 0.4.33 

So as I said, you could add the file here in the mean time.
 
> Where should I add this command?

```shell
$ man flatpak-manifest
/use-git
```

This should give you plenty of answers. (it's not a command either)

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by Murmele** [comment]
bot, build org.tuxemon.Tuxemon

**Comment by Murmele** [comment]
@bitcraft now we have a working build

**Comment by Murmele** [comment]
for me it is working fine

**Pull request closed without merging**
